axelrod/tests/unit/test_prober.py

```diff
@@ -162,3 +161,68 @@ def test_reduction_to_TFT(self):
         test_responses(self, player, opponent, [C], [D], [D])
         test_responses(self, player, opponent, [C, D], [D, C], [C])
         test_responses(self, player, opponent, [C, D], [D, D], [D])
+
+
+class TestRemorsefulProber(TestPlayer):
+
+    name = "Remorseful Prober: 0.1"
+    player = axelrod.RemorsefulProber
+    expected_classifier = {
+        'memory_depth': 2,
+        'stochastic': True,
+        'makes_use_of': set(),
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    def test_strategy(self):
+        "Randomly defects (probes) and always retaliates like tit for tat."
+        self.first_play_test(C)
+
+        player = self.player(0.4)
+        opponent = axelrod.Random()
+        player.history = [C, C]
+        opponent.history = [C, D]
+        self.assertEqual(player.strategy(opponent), D)
+
+    def test_random_defection(self):
+        # Random defection
+        player = self.player(0.4)
+        opponent = axelrod.Random()
+        test_responses(self, player, opponent, [C], [C], [D], random_seed=1)
+
+    def test_remorse(self):
+        """After probing, if opponent retaliates, will offer a C"""
+        player = self.player(0.4)
+        opponent = axelrod.Random()
+
+        random.seed(0)
+        player.history = [C]
+        opponent.history = [C]
+        self.assertEqual(player.strategy(opponent), D)  # Random defection
+        self.assertEqual(player.probing, True)
+
+        player.history = [C, D]
+        opponent.history = [C, D]
+        self.assertEqual(player.strategy(opponent), C)  # Remorse
+        self.assertEqual(player.probing, False)
+
+        player.history = [C, D, C]
+        opponent.history = [C, D, D]
+        self.assertEqual(player.strategy(opponent), D)
+        self.assertEqual(player.probing, False)
+
+    def test_reduction_to_TFT(self):
+        player = self.player(0)
+        opponent = axelrod.Random()
+        test_responses(self, player, opponent, [C], [C], [C], random_seed=1)
+        test_responses(self, player, opponent, [C], [D], [D])
+        test_responses(self, player, opponent, [C, D], [D, C], [C])
```

Comment:
You can use e.g. `attrs = {'probing': False}` with `test_responses` and `responses_test` to check that `self.probing` is appropriately set (also above).